### PR TITLE
[test] Update checks of supported applications on Pilatus

### PR DESCRIPTION
**Pull request by lucamar**
Existing checks of supported applications valid on Eiger can be easily adapted to Pilatus with the following lines (e.g.: `NAMD`):
```
        if self.current_system.name == 'pilatus':
            self.valid_prog_environs = ['cpeIntel']
        else:
            self.valid_prog_environs = ['builtin']
```

### CODE REVIEW

**Comment by lucamar** [comment]
I have marked the pull request as a draft, since I will add the working checks of other supported applications here.

**Comment by lucamar** [comment]
I have provided the updated checks of selected scientific applications on Pilatus: for `LAMMPS` I'd rather use `lmp_mpi` than `lmp_omp`.

**Comment by lucamar** [comment]
NAMD failed the performance check on Piz Daint.

**Comment by lucamar** [comment]
I have updated the reference performance of LAMMPS on Eiger and Pilatus: we might adjust the checks after further testing.

**Review by vkarak — APPROVED** [review]
lgtm

File: cscs-checks/apps/lammps/lammps_check.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
 
 class LAMMPSBaseCheck(rfm.RunOnlyRegressionTest):
     def __init__(self):
-        self.valid_prog_environs = ['builtin']
+        if self.current_system.name == 'pilatus':
+            self.valid_prog_environs = ['cpeGNU']
+        else:
+            self.valid_prog_environs = ['builtin']
```
Comment (ekouts) [inline cscs-checks/apps/lammps/lammps_check.py:17]:
```suggestion
        else:
            self.valid_prog_environs = ['builtin']

```
Please add an extra new line after the if statements

File: cscs-checks/apps/namd/namd_check.py [hunk 2]
```diff
@@ -12,7 +12,10 @@
 class NamdBaseCheck(rfm.RunOnlyRegressionTest):
     def __init__(self, arch, scale, variant):
         self.descr = 'NAMD check (%s, %s)' % (arch, variant)
-        self.valid_prog_environs = ['builtin']
+        if self.current_system.name == 'pilatus':
+            self.valid_prog_environs = ['cpeIntel']
+        else:
+            self.valid_prog_environs = ['builtin']
```
Comment (ekouts) [inline cscs-checks/apps/namd/namd_check.py:18]:
```suggestion
        if self.current_system.name == 'pilatus':
            self.valid_prog_environs = ['cpeIntel']
        else:
            self.valid_prog_environs = ['builtin']

```
Same here

**Review by ekouts — APPROVED** [review]
Lgtm overall, I only have a couple of comments based on the coding style guide of reframe.

**Comment by ekouts** [comment]
Ah sorry, too late 😅

**Comment by vkarak** [comment]
Oh, sorry! You are right btw in your comment, but never mind, we can fix this whenever we touch again the test file.

**Pull request merged**